src/core/listener.c

```diff
@@ -905,6 +1007,26 @@ QuicListenerParamGet(
         Status = QUIC_STATUS_SUCCESS;
         break;
 
+    case QUIC_PARAM_LISTENER_PARTITION_INDEX:
+
+        if (*BufferLength < sizeof(Listener->PartitionIndex)) {
+            *BufferLength = sizeof(Listener->PartitionIndex);
+            return QUIC_STATUS_BUFFER_TOO_SMALL;
+        }
+
+        if (Buffer == NULL) {
+            return QUIC_STATUS_INVALID_PARAMETER;
+        }
+
+        if (!Listener->Partitioned) {
+            return QUIC_STATUS_INVALID_STATE;
+        }
+
+        *BufferLength = sizeof(Listener->PartitionIndex);
+        *(uint16_t*)Buffer = Listener->PartitionIndex;
```

Comment:
There's no guarantee this pointer is aligned so this can crash on ARM platforms. Use `memcpy` instead.

Reply:
I'm not sure I follow - MsQuic can demand callers provide aligned pointers? There are plenty of other precedents with MsQuic where the API expects valid alignment.

Reply:
I see we don't have a consistent stance on this, so I'll drop it for this PR. We should either require callers to use aligned pointers (and document it) or expect callers to provide unaligned pointers and handle it appropriately.